src/backend/aspen/app/views/sample.py

```diff
@@ -230,3 +263,61 @@ def samples():
             results.append(returned_sample_data)
 
         return jsonify({SAMPLE_KEY: results})
+
+
+@application.route("/api/samples/create", methods=["POST"])
+@requires_auth
+def create_sample():
+    with session_scope(application.DATABASE_INTERFACE) as db_session:
+        profile: Mapping[str, str] = session["profile"]
+
+        user: User = (
+            get_usergroup_query(db_session, profile["user_id"])
+            .options(joinedload(User.group))
+            .one()
+        )
+        request_data = request.get_json()
+
+        for data in request_data:
+            data_ok: bool
+            missing_fields: Optional[list[str]]
+            unexpected_fields: Optional[list[str]]
+            data_ok, missing_fields, unexpected_fields = api_utils.check_data(
+                list(data["sample"].keys()),
+                list(data["pathogen_genome"].keys()),
+                SAMPLES_POST_REQUIRED_FIELDS,
+                SAMPLES_POST_OPTIONAL_FIELDS,
+            )
+            if data_ok:
+                sample_args: Mapping[str, Any] = {
```

Comment:
not as important for the first release of this endpoint but would we check here if any of the request data had any of these optional fields in it?

Reply:
oh yeah we have that! so we so check in the `check_data` function if there were any optional fields in the request data.